### PR TITLE
feat: add timeout to read gun

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Avatar/Services/bind.ts [hunk 1]
```diff
@@ -8,18 +8,28 @@ import { delay } from '@masknet/shared-base'
 
 const NFTAvatarGUN = gun2.get(NFT_AVATAR_DB_NAME)
 const READ_GUN_RETRIES = 10
+const READ_GUN_TIMEOUT = 60 * 1000 // timeout 1 minute
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/Services/bind.ts]: 15 sec

File: packages/mask/src/plugins/Avatar/Services/bind.ts [hunk 2]
```diff
@@ -8,18 +8,28 @@ import { delay } from '@masknet/shared-base'
...
 
+function timeout<T>(promise: PromiseLike<T>, time: number): Promise<T | undefined> {
+    return Promise.race([promise, new Promise<undefined>((resolve) => setTimeout(() => resolve(undefined), time))])
+}
+
+async function getDataFromGUN(userId: string) {
+    return (
+        NFTAvatarGUN
+            // @ts-expect-error
+            .get(userId).then!()
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/Services/bind.ts:24]: Does it need to read twice like before?

File: packages/mask/src/plugins/Avatar/Services/bind.ts [hunk 3]
```diff
@@ -4,26 +4,30 @@ import addSeconds from 'date-fns/addSeconds'
...
 
 const cache = new Map<string, [Promise<string | undefined>, number]>()
 
-async function getUserAddressFromGUN(userId: string): Promise<string | undefined> {
-    // eslint-disable-next-line no-plusplus
-    for (let i = 0; i < READ_GUN_RETRIES; i++) {
-        const result = await NFTAvatarGUN
+function timeout<T>(promise: PromiseLike<T>, time: number): Promise<T | undefined> {
+    return Promise.race([promise, new Promise<undefined>((resolve) => setTimeout(() => resolve(undefined), time))])
+}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/Services/bind.ts]:
```suggestion
function timeout<T>(promise: PromiseLike<T>, duration: number): Promise<T | undefined> {
    return Promise.race([promise, new Promise<void>((resolve) => setTimeout(() => resolve(), duration))])
}
```

**Comment by AinuX** [comment]
Checked

File: packages/mask/src/plugins/Avatar/Services/bind.ts [hunk 4]
```diff
@@ -4,26 +4,30 @@ import addSeconds from 'date-fns/addSeconds'
 import { KeyValue } from '@masknet/web3-providers'
 import { NFT_AVATAR_DB_NAME, NFT_AVATAR_DB_NAME_STORAGE } from '../constants'
 import { gun2 } from '../../../network/gun/version.2'
-import { delay } from '@masknet/shared-base'
 
 const NFTAvatarGUN = gun2.get(NFT_AVATAR_DB_NAME)
 const READ_GUN_RETRIES = 10
+const READ_GUN_TIMEOUT = 15 * 1000 // timeout 1 minute
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/Services/bind.ts]:
```suggestion
const READ_GUN_TIMEOUT = 15 * 1000
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**